Exiled.API/Features/Map.cs

```diff
@@ -493,6 +499,33 @@ public static void ChangeUnitColor(int index, string color)
             }
         }
 
+        /// <summary>
+        /// Spawns a live grenade object on the map.
+        /// </summary>
+        /// <param name="position">The <see cref="Vector3"/> to spawn the grenade at.</param>
+        /// <param name="velocity">The <see cref="Vector3"/> directional velocity the grenade should move at.</param>
+        /// <param name="fuseTime">The <see cref="float"/> fuse time of the grenade.</param>
+        /// <param name="grenadeType">The <see cref="GrenadeType"/> of the grenade to spawn.</param>
+        /// <param name="player">The <see cref="Player"/> to count as the thrower of the grenade.</param>
+        /// <returns>The <see cref="Grenade"/> being spawned.</returns>
+        public static Grenade SpawnGrenade(Vector3 position, Vector3? velocity = null, float fuseTime = 3f, GrenadeType grenadeType = GrenadeType.FragGrenade, Player player = null)
+        {
+            if (player == null)
```

Comment:
I would add a check here, to verify whether the grenadeType is valid or not.

Reply:
How can it be invalid? All 3 possible values are valid and are working.

Reply:
I don't think you can set it to `null`

Reply:
Using [this](https://docs.microsoft.com/en-US/dotnet/api/system.enum.tryparse?view=net-5.0).

Reply:
I'm sorry, but I really don't know what you mean. Like, checking the `grenadeType` via `Enum.TryParse()`?? How does parsing the already parsed thing make any sense? I'm probably misunderstood what you meant.

Reply:
You've to parse it and check if it's valid because someone could just do something like `(GrenadeType)999` and it would be counted as a valid enum, even if it doesn't exist.